astropy/coordinates/builtin_frames/cirs_observed_transforms.py

```diff
@@ -40,7 +40,8 @@ def cirs_to_altaz(cirs_coo, altaz_frame):
     xp, yp = get_polar_motion(obstime)
 
     #first set up the astrometry context for CIRS<->AltAz
-    astrom = erfa.apio13(obstime.jd1, obstime.jd2,
+    obstime_utc = obstime if obstime.scale == 'utc' else obstime.utc
```

Comment:
@mhvk - relevant to the above questions, here is the code:
```
    def __getattr__(self, attr):
        """
        Get dynamic attributes to output format or do timescale conversion.
        """
        if attr in self.SCALES and self.scale is not None:
            tm = self.replicate()
            tm._set_scale(attr)
            return tm
```
What about:

- Return `self` if `attr == self.scale`?
- Cache other scales so getting both `jd1` and `jd2` after transforming will be efficient?
Roughly:
```
    def __getattr__(self, attr):
        """
        Get dynamic attributes to output format or do timescale conversion.
        """
        if attr in self.SCALES and self.scale is not None:
            if attr == self.scale:
                tm = self
            elif attr in self._scale_cache:
                tm = self._scale_cache[attr]
            else:
                tm = self.replicate()
                tm._set_scale(attr)
                self._scale_cache[attr] = tm
            return tm
```